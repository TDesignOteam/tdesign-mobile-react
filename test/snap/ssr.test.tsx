import glob from 'glob';
import { vi } from 'vitest';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { IGNORE_ASYNC_EXAMPLE_LIST } from './const';

function runTest() {
  const files = glob.sync('src/**/_example/*.jsx', {
    ignore: IGNORE_ASYNC_EXAMPLE_LIST,
  });

  vi.mock('react-dom', async () => ({
    ...(await vi.importActual('react-dom')),
    createPortal: (node) => node,
  }));

  describe('ssr snapshot test', () => {
    files.forEach((file) => {
      it(`ssr test ${file}`, async () => {
        const demo = await import(`../../${file}`);
        const RealDemoComp = demo.default ? demo.default : demo;
        const html = renderToString(<RealDemoComp />);
        expect(html).toMatchSnapshot();
      });
    });
  });
}

runTest();
