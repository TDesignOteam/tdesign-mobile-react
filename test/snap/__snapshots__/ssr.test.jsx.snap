// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ssr snapshot test > ssr test src/grid/_example/badge.tsx 1`] = `"<div class="t-grid"><div class="t-grid-item" style="flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center"><div class="t-badge"><div class="t-badge__inner t-badge--dot t-badge--has-children t-badge--medium"></div><img class="img-4 img-vertical" src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"/></div><div class="t-grid-item__text" style="padding-left:0"><div class="t-grid-item__title" style="padding-top:8px;margin-bottom:4px"><div class="text-4">标题文字</div></div><div class="t-grid-item__description"></div></div></div><div class="t-grid-item" style="flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center"><div class="t-badge"><div class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium">8</div><img class="img-4 img-vertical" src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"/></div><div class="t-grid-item__text" style="padding-left:0"><div class="t-grid-item__title" style="padding-top:8px;margin-bottom:4px"><div class="text-4">标题文字</div></div><div class="t-grid-item__description"></div></div></div><div class="t-grid-item" style="flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center"><div class="t-badge"><div class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium">new</div><img class="img-4 img-vertical" src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"/></div><div class="t-grid-item__text" style="padding-left:0"><div class="t-grid-item__title" style="padding-top:8px;margin-bottom:4px"><div class="text-4">标题文字</div></div><div class="t-grid-item__description"></div></div></div><div class="t-grid-item" style="flex-basis:25%;flex-direction:column;padding-left:0;padding-right:0;align-items:center;justify-content:center;text-align:center"><div class="t-badge"><div class="t-badge__inner t-badge--circle t-badge--has-children t-badge--medium">···</div><img class="img-4 img-vertical" src="https://tdesign.gtimg.com/mobile/%E5%9B%BE%E7%89%87.png"/></div><div class="t-grid-item__text" style="padding-left:0"><div class="t-grid-item__title" style="padding-top:8px;margin-bottom:4px"><div class="text-4">标题文字</div></div><div class="t-grid-item__description"></div></div></div></div>"`;
